specs/consensus.md

```diff
@@ -456,17 +509,43 @@ validator.pendingRewards = 0
 validator.delegatedCount -= 1
 validator.votingPower -= delegation.votingPower
 
-state.accounts[sender].delegationInfo = delegation
-
-if state.inactiveValidatorSet[delegation.validator].status == ValidatorStatus.Queued ||
-      state.inactiveValidatorSet[delegation.validator].status == ValidatorStatus.Unbonding
-      state.inactiveValidatorSet[delegation.validator].status == ValidatorStatus.Unbonded
+# Update the validator in the linked list by first removing then inserting
+if state.accounts[delegation.validator].status == AccountStatus.ValidatorBonded ||
+      state.accounts[delegation.validator].status == AccountStatus.ValidatorQueued
+    parent = parentFromQueue(delegation.validator)
+    if parent != 0
+        if state.accounts[parent].status == AccountStatus.ValidatorBonded
+            state.activeValidatorSet[parent].next = validator.next
+            validator.next = 0
+        else
+            state.inactiveValidatorSet[parent].next = validator.next
+            validator.next = 0
+    else
+        state.validatorQueueHead = validator.next
+        validator.next = 0
+
+    parent = findFromQueue(validator.votingPower)
+    if parent != 0
+        if state.accounts[parent].status == AccountStatus.ValidatorBonded
+            validator.next = state.activeValidatorSet[parent].next
+            state.activeValidatorSet[parent].next = delegation.validator
+        else
+            validator.next = state.inactiveValidatorSet[parent].next
+            state.inactiveValidatorSet[parent].next = delegation.validator
+    else
+        validator.next = state.validatorQueueHead
+        state.validatorQueueHead = delegation.validator
```

Comment:
Isn't this the exact same pseudo-code as in lines 438 following?

Reply:
Fixed in 33d1f04.